batch-nft-reveal/app/pages/create.tsx

```diff
@@ -17,25 +18,27 @@ function CreatePage(): JSX.Element {
   const [deployedContract, setDeployedContract] = useState<
     DeployedContract | undefined
   >()
+
   const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [error, setError] = useState('')
 
   const onSubmit = useCallback(
     async (args: CreateFormValues) => {
+      setError('')
       setIsLoading(true)
+      let tx
       try {
-        const tx = await deployNFTCollection(
-          args,
-          library?.getSigner(),
-          chainId
-        )
-        setIsLoading(false)
-        setDeployedContract({
-          address: tx.contractAddress,
-          txHash: tx.transactionHash,
-        })
-      } catch (err) {
+        tx = await deployNFTCollection(args, library?.getSigner(), chainId)
+      } catch (ex) {
+        ex.message ? setError(ex.message) : setError(ex)
```

Comment:
Isn't `ex` expected to be an object?

Reply:
No, we can't be sure about that. The errors from Metamask are objects for sure, but the deploy service have other code that can crash too and everything is wrapped in a single try-catch. The best I can do from here is if it's not a Metamask error to show the full message.

Reply:
My concern is setting `any` which could be an object to the error state which is expected to be `string`. Can't we just set a generic error message when that is the case?